functorch/_src/aot_autograd.py

```diff
@@ -472,12 +472,16 @@ def forward(ctx, *deduped_flat_tensor_args):
         def backward(ctx, *flat_args):
             contiguous_args = [t.contiguous() if torch.is_tensor(t) else t for t in flat_args]
             all_args = list(ctx.symints) + list(ctx.saved_tensors) + list(contiguous_args)
+            del contiguous_args
             if CompiledFunction.compiled_bw is None:
+                mode = torch._subclasses.fake_tensor.FakeTensorMode()
+                fake_args = [mode.from_tensor(x) if isinstance(x, torch.Tensor) else x for x in all_args]
                 context = disable_autocast_manager if disable_amp else nullcontext
                 with context(), track_graph_compiling("backward", True):
                     CompiledFunction.compiled_bw = aot_config.bw_compiler(
-                        bw_module, all_args
+                        bw_module, fake_args
```

Comment:
@Chillee how come the backward compiler needs fake args, didn't we already synthesize then when creating the joint graph?

Reply:
What do you mean? We created them, but in the future, it'd be possible for them to be other tensor subclasses.

Also, the other issue is that need to pass symints from the backwards to the forwards pass.

Reply:
It's hard for me to believe this case actually works correctly lol.

Reply:
Like, how do you even fakify an arbitrary torch dispatch tensor subclass